plugins/content_distribution/providers/youtube_api/lib/YoutubeApiDistributionEngine.php

```diff
@@ -257,7 +259,23 @@ protected function doSubmit(KalturaDistributionSubmitJobData $data, KalturaYoutu
 			while (!$ingestedVideo && !feof($handle)) 
 			{
 				$chunk = fread($handle, $chunkSizeBytes);
-				$ingestedVideo = $media->nextChunk($chunk);
+				$numOfTries = 0;
+				while (true) 
+				{
+					try
+					{
+						$ingestedVideo = $media->nextChunk($chunk);
+						break;
+					} catch (Google_IO_Exception $e)
+					{
+						KalturaLog::info("Uploading chunk to youtube failed with the message '".$e->getMessage()."' number of retries ".$numOfTries);
```

Comment:
Instead of implementing retry mechanism yourself, just throw kTemporaryException and the job will be retried.